### ISSUE TITLE
Network Device creation fails due to ACL template

### ISSUE THREAD

**Issue by abudilov** [equinix/terraform-provider-equinix#27]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform Version

Terraform v0.13.5
+ provider developer.equinix.com/terraform/equinix v1.0.0 (master)

### Affected Resource(s)

* equinix_network_device

### Terraform Configuration Files

```tf

resource "equinix_network_device" "network-edge" {
  name               = var.name
  throughput         = var.throughput
  throughput_unit    = var.throughput_unit
  metro_code         = var.metro_code
  type_code          = var.type_code
  package_code       = var.package_code
  notifications      = var.notifications
  hostname           = var.hostname
  term_length        = var.term_length
  account_number     = var.billing_account
  version            = var.sw_version
  core_count         = var.core_count
  # Include redundant device when ha is set to true
  dynamic "secondary_device" {
    for_each = var.ha ? [1] : []
    content { 
      name           = "${var.name}-secondary"
      metro_code     = var.metro_code
      hostname       = var.hostname
      notifications  = var.notifications
      account_number = var.billing_account
    }
  }
```

### Debug Output
```
Error: cannot fetch primary network device ACLs due to Equinix REST API error: Message: "Not Found", HTTPCode: 404, ApplicationErrors: [Code: "", Property: "", Message: "", AdditionalInfo: ""]
```

### Expected Behavior

With the introduction of ACL template in ECX, it is no longer possible to create a network device without specifying the `aclTemplateId` attribute. It is also not clear whether the template should be created first and referenced in the device resource definition or other way around. Looking at the aclTemplates branch I see that the device UUID is an attribute of the ACL template, but as of today you cannot create a device without specifying the template ID.

If the ACL is specified as a list, the ECX backend converts it to a template, but the provider apply still fails with the REST error:
```
Error: cannot fetch primary network device ACLs due to Equinix rest error: httpCode: 404,
```

### Actual Behavior

The apply fails with the REST API error

### Steps to Reproduce

1. `terraform apply`

**Comment by mikouaj** [comment]
@abudilov this should be fixed by #19

**Comment by mikouaj** [comment]
Closing the issue as November release went to prod and ACL template feature was merged.
